smart-contracts/contracts/mixins/MixinGrantKeys.sol

```diff
@@ -35,6 +37,12 @@ contract MixinGrantKeys is
 
       _assignNewTokenId(toKey);
       _recordOwner(recipient, toKey.tokenId);
+      // Only assign the KeyManager for new keys
+      if(keyManagerOf[toKey.tokenId] == address(0)) {
```

Comment:
hmm. I think this needs work.

Reply:
The intent was to prevent the use of `grantKeys` to be able to "take-over" an existing user's key. EG:
you have a valid key. I grant you a new key (which extends yours) but I also set myself as the keyManager for your key.

Reply:
but there's already a check for this above, so I will push a small change to simplify this.